File: app/src/main/java/com/song/deviceinfo/info/RootInfo.java
package com.song.deviceinfo.info;

import android.content.Context;

import com.song.deviceinfo.utils.CommandUtils;
import com.song.deviceinfo.utils.RootUtils;

import java.util.ArrayList;
import java.util.List;

import androidx.core.util.Pair;

/**
 * Created by chensongsong on 2020/7/15.
 */
public class RootInfo {

    public static List<Pair<String, String>> getRootInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        list.add(new Pair<>("Su_v", CommandUtils.execute("su -v")));
        list.add(new Pair<>("RwPaths", RootUtils.existingRWPaths().toString()));
        list.add(new Pair<>("DangerousProperties", RootUtils.existingDangerousProperties().toString()));
        list.add(new Pair<>("RootFiles", RootUtils.existingRootFiles().toString()));
        list.add(new Pair<>("RootPackages", RootUtils.existingRootPackages(context).toString()));
        return list;
    }

}

File: app/src/main/java/com/song/deviceinfo/info/RootInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import androidx.core.util.Pair
import com.song.deviceinfo.utils.CommandUtils.execute
import com.song.deviceinfo.utils.RootUtils.existingDangerousProperties
import com.song.deviceinfo.utils.RootUtils.existingRWPaths
import com.song.deviceinfo.utils.RootUtils.existingRootFiles
import com.song.deviceinfo.utils.RootUtils.existingRootPackages

/**
 * Created by chensongsong on 2020/7/15.
 */
object RootInfo {
    fun getRootInfo(context: Context): List<Pair<String, String>> {
        val list: MutableList<Pair<String, String>> = ArrayList()
        list.add(Pair("Su_v", execute("su -v")))
        list.add(Pair("RwPaths", existingRWPaths().toString()))
        list.add(Pair("DangerousProperties", existingDangerousProperties().toString()))
        list.add(Pair("RootFiles", existingRootFiles().toString()))
        list.add(Pair("RootPackages", existingRootPackages(context).toString()))
        return list
    }
}
